corehq/apps/es/README.rst

```diff
@@ -227,21 +230,19 @@ etc.  This functionality is split into a separate class for a few reasons:
    operations a connection can be used for. The separation in these client
    adapter classes is designed to fit into that model.
 
-The management adapter does not need any special parameters to work with, and
-can be instantiated and used directly:
-
 .. code-block:: python
 
-    adapter = ElasticManageAdapter()
-    adapter.index_create("books")
+    from corehq.apps.es.client import manager
+
+    manager.index_create("books")
     mapping = {"properties": {
         "author": {"type": "text"},
         "title": {"type": "text"},
         "published": {"type": "date"},
     }}
-    adapter.index_put_mapping("books", "book", mapping)
-    adapter.index_refresh("books")
-    adapter.index_delete("books")
+    manager.index_put_mapping("books", "book", mapping)
+    manager.index_refresh("books")
+    manager.index_delete("books")
```

Comment:
I know this didn't change here, but is it true that the arguments passed to these methods are all properties of an adapter? Would it make sense to have them take an adapter instance rather than the primitive arguments that are currently accepted? I think that would make it harder to pass a bad value that could result in incorrect cluster state.

If this idea is accepted it should almost certainly be implemented in a separate PR since it looks like some legacy code would need possibly non-trivial adjustments.

Reply:
I can see `delete`, `index_put_mapping`, `refresh` taking in adapter instances. I was wondering about index creation, would it be better to pass in the adapter instance given that index does not exist?

Reply:
Offlined with @joxl: this will be implemented in a future PR.